src/app/new-page/component/content-animation/content-animation.component.ts

```diff
@@ -0,0 +1,133 @@
+  ready: boolean;
+  anmResource: string;
+  anmViewItem: any;
+  hasEvents: boolean;
+  dir$: Observable<string>;
+  lottieOptions: any;
+
+  constructor(private pageService: PageService) {
+    this.dir$ = this.pageService.pageDir$;
+  }
+
+  ngOnDestroy() {
+    this._unsubscribeAll.next();
+    this._unsubscribeAll.complete();
+  }
+
+  ngOnChanges(changes: SimpleChanges) {
+    for (const propName in changes) {
+      if (changes.hasOwnProperty(propName)) {
+        switch (propName) {
+          case 'item': {
+            if (
+              !changes['item'].previousValue ||
+              changes['item'].currentValue !== changes['item'].previousValue
+            ) {
+              this.ready = false;
+              this.animation = this.item;
+              this.init();
+            }
+          }
+        }
+      }
+    }
+  }
+
+  onAnimationClick(): void {
+    if (this.animation.events) {
+      let action = '';
+      this.animation.events.forEach((event, idx) => {
+        const value = event?.namespace
+          ? `${event.namespace}:${event.name}`
+          : event.name;
+        action += idx ? ` ${value}` : value;
+      });
+      this.pageService.formAction(action);
+    }
+  }
+
+  onAnimationCreated(anim: any) {
+    this.anmViewItem = anim;
+  }
+
+  private init(): void {
+    // TODO
+    // Need to update to e31_1 when we release a new Shared Parser.
+    const resource = {
+      name: (this.animation as any).e31_1 || ''
+    };
+    this.anmResource = this.pageService.getAnimationUrl(resource.name || '');
+    if (
+      this.anmResource === resource.name &&
+      !this.anmResource.includes('http')
+    ) {
+      this.anmResource = this.pageService.findAttachment(resource.name) || '';
+    }
+
+    if (this.anmResource) {
+      this.lottieOptions = {
+        path: this.anmResource,
+        loop: !!this.animation.loop,
+        autoplay: !!this.animation.autoPlay
+      };
+    }
+
+    this.hasEvents = !!this.animation.events;
+
+    if (!!this.animation.playListeners || !!this.animation.stopListeners) {
+      this.awaitAnimEvent();
+    }
+    this.ready = true;
+  }
+
+  private awaitAnimEvent(): void {
+    this.pageService.contentEvent$
+      .pipe(takeUntil(this._unsubscribeAll))
+      .subscribe((event) => {
+        if (!this.anmViewItem) return;
+        const playListeners = this.animation.playListeners.filter(
+          (listener) => listener.name === event
+        ).length;
+        const stopListeners = this.animation.stopListeners.filter(
+          (listener) => listener.name === event
+        ).length;
+        const events = this.animation.events.filter(
+          (e) => e.name === event
+        ).length;
+
+        if (playListeners || stopListeners) {
+          if (events) {
+            this.anmViewItem.togglePause();
+          } else if (playListeners) {
+            this.anmViewItem.play();
+          } else if (stopListeners) {
+            this.anmViewItem.stop();
```

Comment:
I think you would want to `pause()` the animation instead of `.stop()`, if lottie supports that differentiation on the web

Reply:
here is how I'm doing play/stop listeners on Android: https://github.com/CruGlobal/godtools-android/blob/develop/ui/base-tool/src/main/kotlin/org/cru/godtools/base/tool/ui/controller/AnimationController.kt#L38-L45

Reply:
Lottie supports pause and stop, so I've changed it to pause. I didn't think there was a difference. However, Lottie Web doesn't support `resume`.

Reply:
if I recall, pause maintains the current playback position, stop resets it to the beginning